test/e2e/gardenadm/hightouch/gardenadm.go

```diff
@@ -185,6 +185,14 @@ var _ = Describe("gardenadm high-touch scenario tests", Label("gardenadm", "high
 			}).Should(HaveKeyWithValue("injected-by", "provider-local"))
 		}, SpecTimeout(time.Minute))
 
+		It("should find the cloud provider secret", func(ctx SpecContext) {
```

Comment:
Confusing, as mentioned above 😅

Reply:
Same as above, i.e. I removed the check for the high touch test case.

@timebertt Please add a corresponding check once the cloud provider secret is deployed in the medium touch case.

Reply:
I refrained from asking earlier, but now it comes up again: do you want to add deployment of the `cloudprovider` secret to the `gardenadm bootstrap` flow in this PR, too? Considering these questions, it might be reasonable to tackle both flows in one rush.

Reply:
Ok, let me try.

Reply:
I added the `cloudprovider` secret now also to the `gardenadm bootstrap` scenario. This required major rework with regards to the manifest management. Now, we can easily customise/diverge between high and medium touch scenarios.
WDYT?